lib/shopify_app/carrier_services_manager.rb

```diff
@@ -0,0 +1,55 @@
+module ShopifyApp
+  class CarrierServicesManager
+    class CreationFailed < StandardError; end
+
+    def self.queue(shop_domain, shop_token)
+      ShopifyApp::CarrierServicesManagerJob.perform_later(shop_domain: shop_domain, shop_token: shop_token)
+    end
+
+    def recreate_carrier_services!
+      destroy_carrier_services
+      create_carrier_services
+    end
+
+    def create_carrier_services
+      return unless required_carrier_services.present?
+
+      required_carrier_services.each do |carrier_service|
```

Comment:
if `required_carrier_services` defaulted to `[]` we wouldn't need to check presence and could just have the loop which would be a no-op